pkg/controller/jobframework/reconciler.go

```diff
@@ -292,27 +291,22 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	isTopLevelJob := true
-	objectOwner := metav1.GetControllerOf(object)
-	if objectOwner != nil && IsOwnerManagedByKueue(objectOwner) {
-		isTopLevelJob = false
+	managedOwnersChain, err := r.getManagedOwnersChain(ctx, job)
+	if err != nil {
+		return ctrl.Result{}, err
 	}
+	isTopLevelJob := len(managedOwnersChain) == 0
+	ancestorJob := FindAncestorJobManagedByKueue(managedOwnersChain)
 
 	// when manageJobsWithoutQueueName is disabled we only reconcile jobs that either
 	// have a queue-name label or have a kueue-managed ancestor that has a queue-name label.
 	if !r.manageJobsWithoutQueueName && QueueName(job) == "" {
 		if isTopLevelJob {
-			log.V(3).Info("queue-name label is not set, ignoring the job", "queueName", QueueName(job))
+			log.V(3).Info("queue-name label is not set, ignoring the job")
```

Comment:
Why is this changed?

Reply:
The queue name is an empty string, so this isn't needed.